### PR TITLE
BFD-3161: Deploy EFT Terraservice during regular deployment

### PR DESCRIPTION
**Pull request by malessi**
**JIRA Ticket:**
[BFD-3161](https://jira.cms.gov/browse/BFD-3161)

**User Story or Bug Summary:**

As a BFD Engineer, I want the `eft` Terraservice to be deployed regularly so that the infrastructure generated from the service is always up-to-date and so that I do not need to manually deploy it myself.

---

### What Does This PR Do?

This PR:
- adds new stages to the Deployment Pipeline that deploys the `eft` Terraservice in `test` and `prod` during typical deployment.
- a new Github Actions Workflow: `Build Docker Images`. 
  - as of this PR this Workflow only builds the BFD EFT Outbound lambda Docker Image, but it is written such that multiple images can be specified and built in-parallel (using the `matrix` strategy). The corresponding Jenkins pipeline that _also_ builds this image has not been removed -- this should be done in future work (i.e. BFD-2681 or similar). 
- `Build Release` Workflow has been updated to invoke the new `Build Docker Images` Workflow such that the image will be built on each release. 
- the `Build Release` Workflow has also been refactored into multiple sub-jobs such that a GitHub Release is only created if the Maven release succeeds _and_ the Docker images are built (see [this mock run for an example](https://github.com/CMSgov/beneficiary-fhir-data/actions/runs/7426869379)).

Additionally, a number of fixes have been made to the `eft` Terraservice and its corresponding `base` configuration so that it can be `apply`'d properly.

This PR has been verified by:

- `terraform apply`ing `eft` in `prod`, **verifying that** the `eft` Terraservice applies as expected without error
- running a [modified `Build Release` workflow](https://github.com/CMSgov/beneficiary-fhir-data/actions/runs/7426869379) (with jobs/stages that build and generate a GH release disabled), **verifying that** the workflow succeeded and that the BFD EFT Outbound Docker image was generated and pushed to the correct ECR repository

### What Should Reviewers Watch For?

If you're reviewing this PR, please check for these things in particular:

- Verify all PR security questions and checklists have been completed and addressed.

### What Security Implications Does This PR Have?

Submitters should complete the following questionnaire:

- If the answer to any of the questions below is **Yes**, then **you must** supply a link to the associated Security Impact Assessment (SIA), security checklist, or other similar document in Confluence here: **N/A**

  - Does this PR add any new software dependencies?
    - [ ] Yes
    - [x] No
  - Does this PR modify or invalidate any of our security controls?
    - [ ] Yes
    - [x] No
  - Does this PR store or transmit data that was not stored or transmitted before?
    - [ ] Yes
    - [x] No

- If the answer to any of the questions below is **Yes**, then please add @StewGoin as a reviewer, and note that this PR **should not be merged** unless/until he also approves it.
  - Do you think this PR requires additional review of its security implications for other reasons?
    - [ ] Yes
    - [x] No

### What Needs to Be Merged and Deployed Before this PR?

This PR cannot be either merged or deployed until the following prerequisite changes have been fully deployed:

- N/A

### Submitter Checklist

I have gone through and verified that...:

- [x] I have named this PR and branch so they are [automatically linked](https://confluence.atlassian.com/adminjiracloud/integrating-with-development-tools-776636216.html) to the (most) relevant Jira issue. Ie: `BFD-123: Adds foo`
- [x] This PR is reasonably limited in scope, to help ensure that:
  1. It doesn't unnecessarily tie a bunch of disparate features, fixes, refactorings, etc. together.
  2. There isn't too much of a burden on reviewers.
  3. Any problems it causes have a small "blast radius".
  4. It'll be easier to rollback if that becomes necessary.
- [x] This PR includes any required documentation changes, including `README` updates and changelog / release notes entries.
- [x] The data dictionary has been updated with any field mapping changes, if any were made.
- [x] All new and modified code is appropriately commented, such that the what and why of its design would be reasonably clear to engineers, preferably ones unfamiliar with the project.
- [x] All tech debt and/or shortcomings introduced by this PR are detailed in `TODO` and/or `FIXME` comments, which include a JIRA ticket ID for any items that require urgent attention.
- [x] Reviews are requested from both:
  - At least two other engineers on this project, at least one of whom is a senior engineer or owns the relevant component(s) here.
  - Any relevant engineers on other projects (e.g. DC GEO, BB2, etc.).
- [x] Any deviations from the other policies in the [DASG Engineering Standards](https://github.com/CMSgov/cms-oeda-dasg/blob/master/policies/engineering_standards.md) are specifically called out in this PR, above.
  - Please review the standards every few months to ensure you're familiar with them.

### CODE REVIEW

File: .github/workflows/build-docker-images.yml [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+        if: ${{ contains(inputs.versionTag, '-') }}
+        uses: docker/build-push-action@v5
+        with:
+          file: ${{ matrix.image.dockerfile }}
+          context: ${{ matrix.image.contextDir }}
+          push: true
+          tags: |
+            ${{ env.ECR_REPOSITORY_NAMESPACE }}/${{ matrix.image.name }}:${{ inputs.versionTag }}
+        env:
+          DOCKER_BUILDKIT: 1
```
Comment (malessi) [inline .github/workflows/build-docker-images.yml:124]: I am unsure how to reduce this duplication. If anyone understands GitHub Actions better than I, please let me know how to make this better.

**Review by brandoncruz3 — APPROVED** [review]

File: .github/workflows/build-docker-images.yml (hunk 1 above)
Comment (malessi) [reply .github/workflows/build-docker-images.yml]: This has been resolved by using an intermediate step to generate the list of tags conditionally, then passing that list to the `docker/build-push-action` via a step output.

File: ops/terraform/services/base/values/prod.yaml
Comment (malessi) [inline ops/terraform/services/base/values/prod.yaml:1]: There are a lot of changes to this file, but I only added a new parameter (really, missing), `eft_user_home_dir`. I'm not sure why other parameters were updated, but their actual value did not change and I verified that `terraform apply`ing `base` in `prod` results in no changes to the corresponding parameters.

**Review by brandoncruz3 — APPROVED** [review]
🚀

File: Jenkinsfile [hunk 2]
```diff
@@ -539,6 +553,20 @@ try {
 				}
 			}
 
+			stage('Deploy EFT to PROD') {
```
Comment (oeda-dasg-tep) [inline Jenkinsfile:556]: Does this mean you wouldn't deploy to the prod-sbx environment?

**Pull request merged**